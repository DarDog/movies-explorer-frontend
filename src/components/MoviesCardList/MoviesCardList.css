.movies {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;

}

.movies__list {
  margin: 0;
  padding: 0;
  list-style: none;
  width: 100%;
}

.main__movies {
  width: 100%;
  padding: 70px 70px 10px;
  box-sizing: border-box;
}
