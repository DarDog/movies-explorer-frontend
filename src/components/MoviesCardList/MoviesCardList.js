import React from 'react';
import './MoviesCardList.css'
import MoviesCard from "../MoviesCard/MoviesCard";

const MoviesCardList = ({ movies = [] }) => {
  return (
    <section className='movies main__movies'>
      <ul className='movies__list'>
        {
          movies.map(movie => {
            return (
              <MoviesCard key={movie._id} movie={movie} />
            );
          })
        }
      </ul>
    </section>
  );
}

export default MoviesCardList
