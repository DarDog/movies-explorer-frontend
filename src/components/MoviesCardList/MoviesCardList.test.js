import React from 'react';
import MoviesCardList from "./MoviesCardList";
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom';
import movies from "../../utils/preparedFilms";

describe('Компонент MoviesCardList', () => {
  it('Рендерется', () => {
    render(<MoviesCardList movies={movies}/>)

    expect(screen.getByRole('list')).toBeInTheDocument();
  });
  it('Соответствует snapshot', () => {
    const component = render(
      <MoviesCardList movies={movies} />
    )

    expect(component).toMatchSnapshot();
  });
})
