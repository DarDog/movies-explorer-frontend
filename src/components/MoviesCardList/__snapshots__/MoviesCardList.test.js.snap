// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент MoviesCardList Соответствует snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="movies main__movies"
      >
        <ul
          class="movies__list"
        >
          <li
            class="movies__item"
          >
            <div
              class="movies__info"
            >
              <h2
                class="movies__title"
              >
                33 слова о дизайне
              </h2>
              <p
                class="movies__duration"
              >
                1
                ч 
                42
                м
              </p>
              <button
                class="movies__like movies__like_active"
              />
            </div>
            <img
              alt="Заглушка"
              class="movies__poster"
              src="https://via.placeholder.com/220x122"
            />
          </li>
          <li
            class="movies__item"
          >
            <div
              class="movies__info"
            >
              <h2
                class="movies__title"
              >
                Киноальманах «100 лет дизайна»
              </h2>
              <p
                class="movies__duration"
              >
                1
                ч 
                42
                м
              </p>
              <button
                class="movies__like movies__like_active"
              />
            </div>
            <img
              alt="Заглушка"
              class="movies__poster"
              src="https://via.placeholder.com/440x244"
            />
          </li>
          <li
            class="movies__item"
          >
            <div
              class="movies__info"
            >
              <h2
                class="movies__title"
              >
                В погоне за Бенкси
              </h2>
              <p
                class="movies__duration"
              >
                1
                ч 
                42
                м
              </p>
              <button
                class="movies__like false"
              />
            </div>
            <img
              alt="Заглушка"
              class="movies__poster"
              src="https://via.placeholder.com/1280x900"
            />
          </li>
          <li
            class="movies__item"
          >
            <div
              class="movies__info"
            >
              <h2
                class="movies__title"
              >
                Баския: Взрыв реальности
              </h2>
              <p
                class="movies__duration"
              >
                1
                ч 
                42
                м
              </p>
              <button
                class="movies__like false"
              />
            </div>
            <img
              alt="Заглушка"
              class="movies__poster"
              src="https://via.placeholder.com/600x600"
            />
          </li>
          <li
            class="movies__item"
          >
            <div
              class="movies__info"
            >
              <h2
                class="movies__title"
              >
                Бег это свобода
              </h2>
              <p
                class="movies__duration"
              >
                1
                ч 
                42
                м
              </p>
              <button
                class="movies__like movies__like_active"
              />
            </div>
            <img
              alt="Заглушка"
              class="movies__poster"
              src="https://via.placeholder.com/110x61"
            />
          </li>
          <li
            class="movies__item"
          >
            <div
              class="movies__info"
            >
              <h2
                class="movies__title"
              >
                Книготорговцы
              </h2>
              <p
                class="movies__duration"
              >
                1
                ч 
                42
                м
              </p>
              <button
                class="movies__like false"
              />
            </div>
            <img
              alt="Заглушка"
              class="movies__poster"
              src="https://via.placeholder.com/122x220"
            />
          </li>
          <li
            class="movies__item"
          >
            <div
              class="movies__info"
            >
              <h2
                class="movies__title"
              >
                Когда я думаю о Германии ночью
              </h2>
              <p
                class="movies__duration"
              >
                1
                ч 
                42
                м
              </p>
              <button
                class="movies__like false"
              />
            </div>
            <img
              alt="Заглушка"
              class="movies__poster"
              src="https://via.placeholder.com/244x440"
            />
          </li>
        </ul>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="movies main__movies"
    >
      <ul
        class="movies__list"
      >
        <li
          class="movies__item"
        >
          <div
            class="movies__info"
          >
            <h2
              class="movies__title"
            >
              33 слова о дизайне
            </h2>
            <p
              class="movies__duration"
            >
              1
              ч 
              42
              м
            </p>
            <button
              class="movies__like movies__like_active"
            />
          </div>
          <img
            alt="Заглушка"
            class="movies__poster"
            src="https://via.placeholder.com/220x122"
          />
        </li>
        <li
          class="movies__item"
        >
          <div
            class="movies__info"
          >
            <h2
              class="movies__title"
            >
              Киноальманах «100 лет дизайна»
            </h2>
            <p
              class="movies__duration"
            >
              1
              ч 
              42
              м
            </p>
            <button
              class="movies__like movies__like_active"
            />
          </div>
          <img
            alt="Заглушка"
            class="movies__poster"
            src="https://via.placeholder.com/440x244"
          />
        </li>
        <li
          class="movies__item"
        >
          <div
            class="movies__info"
          >
            <h2
              class="movies__title"
            >
              В погоне за Бенкси
            </h2>
            <p
              class="movies__duration"
            >
              1
              ч 
              42
              м
            </p>
            <button
              class="movies__like false"
            />
          </div>
          <img
            alt="Заглушка"
            class="movies__poster"
            src="https://via.placeholder.com/1280x900"
          />
        </li>
        <li
          class="movies__item"
        >
          <div
            class="movies__info"
          >
            <h2
              class="movies__title"
            >
              Баския: Взрыв реальности
            </h2>
            <p
              class="movies__duration"
            >
              1
              ч 
              42
              м
            </p>
            <button
              class="movies__like false"
            />
          </div>
          <img
            alt="Заглушка"
            class="movies__poster"
            src="https://via.placeholder.com/600x600"
          />
        </li>
        <li
          class="movies__item"
        >
          <div
            class="movies__info"
          >
            <h2
              class="movies__title"
            >
              Бег это свобода
            </h2>
            <p
              class="movies__duration"
            >
              1
              ч 
              42
              м
            </p>
            <button
              class="movies__like movies__like_active"
            />
          </div>
          <img
            alt="Заглушка"
            class="movies__poster"
            src="https://via.placeholder.com/110x61"
          />
        </li>
        <li
          class="movies__item"
        >
          <div
            class="movies__info"
          >
            <h2
              class="movies__title"
            >
              Книготорговцы
            </h2>
            <p
              class="movies__duration"
            >
              1
              ч 
              42
              м
            </p>
            <button
              class="movies__like false"
            />
          </div>
          <img
            alt="Заглушка"
            class="movies__poster"
            src="https://via.placeholder.com/122x220"
          />
        </li>
        <li
          class="movies__item"
        >
          <div
            class="movies__info"
          >
            <h2
              class="movies__title"
            >
              Когда я думаю о Германии ночью
            </h2>
            <p
              class="movies__duration"
            >
              1
              ч 
              42
              м
            </p>
            <button
              class="movies__like false"
            />
          </div>
          <img
            alt="Заглушка"
            class="movies__poster"
            src="https://via.placeholder.com/244x440"
          />
        </li>
      </ul>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
